src/coefficients/stdfunction_coefficient.hpp

```diff
@@ -290,11 +310,13 @@ class TransformedScalarCoefficient : public mfem::Coefficient {
    * @brief The first scalar coefficient in the transformation
    */
   std::shared_ptr<mfem::Coefficient> s1_;
+  std::unique_ptr<mfem::Coefficient> u1_;
```

Comment:
While I definitely appreciate the avoidance of `shared_ptr`s here, I'm wondering if it's worth leaving everything as `shared_ptr`s until we develop a better way of managing coefficient ownership in #252.  `unique_ptr`s can be "promoted" via a converting constructor, so there wouldn't need to be any changes to the `TransformedCoef` interface.